### PR TITLE
teach EVAL about nqp

### PR DESCRIPTION
**Pull request by cygx**
The language is called lower-case `nqp` as that's what `nqp::getcomp` expects.

~~It might be a good idea to add another candidate that dispatches on `:lang<NQP>` for consistency with with `use :from<NQP>`...~~

For consistency, the spelling `NQP` is also allowed

### CODE REVIEW

**Comment by lizmat** [comment]
Letting this one for jnthn to look into

File: src/core/control.pm [hunk 1]
```diff
@@ -215,6 +215,10 @@ proto sub EVAL(Cool $code, Str() :$lang = 'perl6', PseudoStash :$context, *%n) {
     nqp::forceouterctx(nqp::getattr($compiled, ForeignCode, '$!do'), $eval_ctx);
     $compiled();
 }
+multi sub EVAL(Cool $code, Str() :$lang! where 'nqp', *%_) {
```
Comment (vendethiel) [inline src/core/control.pm]: the where should probably follow the sub below with `$lang // ''`? or maybe it should be changed to use `:$lang!`?

**Comment by cygx** [comment]
Turns out there is the potential for fallout: Calling `EVAL` at `BEGIN`-time can mess up precompilation.

I have not had any issues using it at run-time.

**Comment by JJ** [comment]
Can you fix conflicts before we go any further?

**Comment by lizmat** [comment]
Looking at the code, it looks like the EVAL would be recursing infinitely?

**Pull request closed without merging**